lib/web/ui/integration.go

```diff
@@ -56,6 +64,10 @@ func (r *Integration) CheckAndSetDefaults() error {
 		return trace.BadParameter("missing awsoidc.roleArn field")
 	}
 
+	if r.AWSOIDC.IssuerS3Bucket != "" && r.AWSOIDC.IssuerS3Prefix == "" {
+		return trace.BadParameter("prefix is required for using s3 buckets")
+	}
+
```

Comment:
It should be something like a XOR condition? So both should be empty or set?
```suggestion
         // Either both should be set or both should be empty.
	if r.AWSOIDC.IssuerS3Bucket == "" != r.AWSOIDC.IssuerS3Prefix == "" {
		return trace.BadParameter("both s3 prefix and bucket should be set")
	}

```

Reply:
I think it's better to always require both